File: test/shire/workspace_test.exs
defmodule Shire.WorkspaceTest do
  use ExUnit.Case, async: false

  import Mox

  alias Shire.Workspace

  @project_id "test-project-id"

  setup do
    Mox.set_mox_global()
    stub(Shire.VirtualMachineMock, :workspace_root, fn _project_id -> "/workspace" end)
    stub(Shire.VirtualMachineMock, :vm_status, fn _project_id -> :running end)
    :ok
  end

  describe "root/1" do
    test "returns workspace root for project" do
      assert Workspace.root(@project_id) == "/workspace"
    end
  end

  describe "agents_dir/1" do
    test "returns agents directory under workspace root" do
      assert Workspace.agents_dir(@project_id) == "/workspace/agents"
    end
  end

  describe "agent_dir/2" do
    test "returns agent directory under agents dir" do
      assert Workspace.agent_dir(@project_id, "agent-123") == "/workspace/agents/agent-123"
    end
  end

  describe "shared_dir/1" do
    test "returns shared directory" do
      assert Workspace.shared_dir(@project_id) == "/workspace/shared"
    end
  end

  describe "env_path/1" do
    test "returns .env path" do
      assert Workspace.env_path(@project_id) == "/workspace/.env"
    end
  end

  describe "scripts_dir/1" do
    test "returns .scripts directory" do
      assert Workspace.scripts_dir(@project_id) == "/workspace/.scripts"
    end
  end

  describe "script_path/2" do
    test "returns path to a named script" do
      assert Workspace.script_path(@project_id, "setup.sh") == "/workspace/.scripts/setup.sh"
    end
  end

  describe "runner_dir/1" do
    test "returns .runner directory" do
      assert Workspace.runner_dir(@project_id) == "/workspace/.runner"
    end
  end

  describe "peers_path/1" do
    test "returns peers.yaml path" do
      assert Workspace.peers_path(@project_id) == "/workspace/peers.yaml"
    end
  end

  describe "project_doc_path/1" do
    test "returns PROJECT.md path" do
      assert Workspace.project_doc_path(@project_id) == "/workspace/PROJECT.md"
    end
  end
end

File: src/services/workspace.test.ts
import { describe, it, expect } from "bun:test";
import * as workspace from "./workspace";

const PROJECT_ID = "test-project-id";

describe("workspace paths", () => {
  it("root returns project directory", () => {
    const r = workspace.root(PROJECT_ID);
    expect(r).toContain(PROJECT_ID);
    expect(r.endsWith(PROJECT_ID)).toBe(true);
  });

  it("agentsDir is under root", () => {
    expect(workspace.agentsDir(PROJECT_ID)).toBe(workspace.root(PROJECT_ID) + "/agents");
  });

  it("agentDir includes agent id", () => {
    expect(workspace.agentDir(PROJECT_ID, "agent-123")).toBe(
      workspace.root(PROJECT_ID) + "/agents/agent-123",
    );
  });

  it("sharedDir returns shared directory", () => {
    expect(workspace.sharedDir(PROJECT_ID)).toBe(workspace.root(PROJECT_ID) + "/shared");
  });

  it("peersPath returns peers.yaml path", () => {
    expect(workspace.peersPath(PROJECT_ID)).toBe(workspace.root(PROJECT_ID) + "/peers.yaml");
  });

  it("projectDocPath returns PROJECT.md path", () => {
    expect(workspace.projectDocPath(PROJECT_ID)).toBe(workspace.root(PROJECT_ID) + "/PROJECT.md");
  });

  it("attachmentPath constructs full path", () => {
    const path = workspace.attachmentPath(PROJECT_ID, "agent-1", "att-1", "file.pdf");
    expect(path).toContain("agents/agent-1/attachments/att-1/file.pdf");
  });
});
